### PR TITLE
[#IOPID-1771] Remove all the APIs migrated to the Session Manager

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### Motivation and Context
The new Session Manager component will handle all the session related functionalities that was provided by the `io-backend`. This functions, APIs, specs and type are removed by this PR.
The removed code makes the `io-backend` startup time faster and remove external dependency needed before the service become healty.

#### How Has This Been Tested?
local build and unit test

The local start with docker of the backend show the reduction of the startup time from `1019 ms` to `453 ms`.

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/config.ts
Comment (gquadrati) [inline src/config.ts:1]: We can remove the variables related to unique email enforcement, and all related functionalities.
I think we could remove `TOKEN_DURATION_IN_SECONDS` var too, by cleaning the RedisSessionStorage from set/update user functionalities

File: src/services/__tests__/lollipopService.test.ts [hunk 1]
```diff
@@ -1,67 +1,27 @@
...
-const mockActivatePubKey = jest.fn();
-const mockLollipopApiClient = {
-  ping: jest.fn(),
-  activatePubKey: mockActivatePubKey,
-  generateLCParams: jest.fn(),
-  reservePubKey: jest.fn()
-} as ReturnType<LollipopApiClient>;
+const mockTelemetryClient = {
+  trackEvent: jest.fn(),
+} as unknown as appInsights.TelemetryClient;
```
Comment (gquadrati) [inline src/services/__tests__/lollipopService.test.ts:22]: The appInsights telemetryClient is not used anymore, we can remove it

File: src/controllers/authenticationController.ts [hunk 2]
```diff
@@ -119,820 +36,32 @@ export const AGE_LIMIT = 14;
 
 export type OnUserLogin = (data: UserLoginParams) => TE.TaskEither<Error, true>;
```
Comment (gquadrati) [inline src/controllers/authenticationController.ts:37]: We can remove it

File: src/controllers/authenticationController.ts
Comment (gquadrati) [inline src/controllers/authenticationController.ts:1]: We can remove all constant related to age limit functionality and session token lenght

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/utils/fastLogin.ts
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/utils/fastLogin.ts:1]: removed in b60ab4f4

File: src/utils/ioLoginUriScheme.ts
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/utils/ioLoginUriScheme.ts:1]: removed in 2c4ae6ed

File: src/types/user.ts
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/types/user.ts:1]: removed in https://github.com/pagopa/io-backend/commit/ee10e3c63f2af2cf536d3b7206ffd10228afde4b

File: src/config.ts
Comment (BurnedMarshal) [reply src/config.ts:1]: unique email enforcement variables are used into the update profile API, cannot be removed.
Removed set/update func in RedisSessionStorage and  `TOKEN_DURATION_IN_SECONDS` env  in 2f76c12c

File: src/services/__tests__/lollipopService.test.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/services/__tests__/lollipopService.test.ts]: removed in 33ceeb92

File: src/controllers/authenticationController.ts
Comment (BurnedMarshal) [reply src/controllers/authenticationController.ts:1]: Removed in a54714d4

**Review by arcogabbo — CHANGES_REQUESTED** [review]
fenomenal work!

PS: `createNewProfile` method in `src/utils/profile.ts` can be removed

File: package.json [hunk 3]
```diff
@@ -126,8 +120,7 @@
     "ts-pattern": "^4.0.3",
     "ulid": "^2.2.2",
     "validator": "^10.8.0",
-    "winston": "^3.3.3",
-    "xmldom": "^0.6.0"
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply package.json:130]: Fixed in 1d0aae8b

File: src/services/redisSessionStorage.ts
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts:1]: Remove in e23d92f5

File: helm/values-prod01.yaml
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply helm/values-prod01.yaml:1]: Remove in e23d92f5

File: src/services/ISessionStorage.ts
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/services/ISessionStorage.ts:1]: done in e23d92f5

**Review by arcogabbo — APPROVED** [review]

**Review by gquadrati — APPROVED** [review]
🚀

**Pull request merged**